### DISCUSSION TITLE
Debian 11 can not access Sonoff dongle Error: Error while opening serialport

### DISCUSSION THREAD

**Discussion by tonbor** [Koenkk/zigbee2mqtt#16436] [Devices]
Zigbee2MQTT:info  2023-01-31 11:42:05: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2023-01-31.11-42-05' filename: log.txt
Zigbee2MQTT:info  2023-01-31 11:42:05: Starting Zigbee2MQTT version 1.29.2 (commit #14021395)
Zigbee2MQTT:info  2023-01-31 11:42:05: Starting zigbee-herdsman (0.14.83-hotfix.0)
Zigbee2MQTT:error 2023-01-31 11:42:07: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2023-01-31 11:42:07: Failed to start zigbee
Zigbee2MQTT:error 2023-01-31 11:42:07: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2023-01-31 11:42:07: Exiting...
Zigbee2MQTT:error 2023-01-31 11:42:07: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
    at SerialPort.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/znp/znp.ts:146:28)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/@serialport/stream/dist/index.js:76:22)
    at /opt/zigbee2mqtt/node_modules/@serialport/stream/dist/index.js:112:18

rob@NUC5-Rob:/opt/zigbee2mqtt$ test -w /dev/ttyUSB0 && echo success || echo failure
success

No access
rob@NUC5-Rob:/opt/zigbee2mqtt$ ls -l /proc/[0-9]/fd/ |grep /dev/ttyUSB0
ls: kan map '/proc/1/fd/' niet openen: Toegang geweigerd
ls: kan map '/proc/2/fd/' niet openen: Toegang geweigerd
ls: kan map '/proc/3/fd/' niet openen: Toegang geweigerd
ls: kan map '/proc/4/fd/' niet openen: Toegang geweigerd
ls: kan map '/proc/6/fd/' niet openen: Toegang geweigerd
ls: kan map '/proc/8/fd/' niet openen: Toegang geweigerd
ls: kan map '/proc/9/fd/' niet openen: Toegang geweigerd
With sudo Access
rob@NUC5-Rob:/opt/zigbee2mqtt$ sudo ls -l /proc/[0-9]/fd/ |grep /dev/ttyUSB0

rob@NUC5-Rob:/opt/zigbee2mqtt$ ls -l /dev/serial/by-id
totaal 0
lrwxrwxrwx 1 root root 13 31 jan 11:24 usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_ce24f433c4c9eb11bc9e914f1d69213e-if00-port0 -> ../../ttyUSB0

https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#method-1-give-your-user-permissions-on-every-reboot
[No solution]

Any help, take me hours brings me no further

**Comment by tonbor** [comment]
No help?????

**Comment by lux73** [comment]
i assume this is a new fresh install? if so try

`apt remove brltty --purge`

this package is prevent z2m to access to your Stick

**and please be patient next time** - pushing your Problem after 10hrs waiting is not really useful!

**Comment by tonbor** [comment]
rob@NUC5-Rob:~$ sudo apt remove brltty --purge

Pakket 'brltty' is niet geïnstalleerd, en wordt dus niet verwijderd -> not installed so not removed.
No luck

**Reply by lux73** [reply]
it was worth a try - on my fresh Debian install few Month ago this was my show-stopper

i was unable to start z2m without removing the mentioned package

**Comment by tonbor** [comment]
Strange, npm start cannot access the port, zigbee to domoticz runs with no problem

**Reply by lux73** [reply]
is your User Member of the `dialout` Group?

https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#method-2-add-your-user-to-specific-groups

**Comment by tonbor** [comment]
Yes

**Comment by lux73** [comment]
you have to add your User to the Group `tty` too - i have installed a fresh Debian VM to reproduce this

`sudo usermod -a -G tty YOURUSER`

**Reply by lux73** [reply]
make sure the z2m and all subdirs have the correct User rights

`chmod -R YOURUSER /opt/zigbee2mqtt`

you have to reapeat this eventually after every `git pull` (update)

**Comment by tonbor** [comment]
ok have to try again, all these things I did, but it seems these rights do not hold for ever

**Reply by lux73** [reply]
plain Debian install or any Docker installation?

**Comment by tonbor** [comment]
plain Debian 11

**Comment by lux73** [comment]
okay lets Start from the Beginning:

`lsusb` - shows your plugged Stick?

something similar to this `Bus 001 Device 002: ID 10c4:ea60 Silicon Labs CP210x UART Bridge`

`ls -al /dev/ttyU*` shows your Stick like

`crw-rw---- 1 root dialout 188, 0  5. Feb 11:31 /dev/ttyUSB0`

show the Output from

`groups YOURUSER` - mine User is called "zigbee"

`zigbee : zigbee tty dialout`

please show all your Output from my posted cmd's

**Reply by tonbor** [reply]
Well I gave up on zigbee2mqtt, I started Zigbee to domoticz because is is working. But when I saw Zigbee to domoticz I realized zigbee2mqtt has far more possibility, so I try it one more time, I disabled domoticz and this are the results you asked for

rob@NUC5-Rob:~$ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 005: ID 8087:0a2a Intel Corp. Bluetooth wireless interface
Bus 001 Device 004: ID 05e3:0610 Genesys Logic, Inc. Hub
Bus 001 Device 003: ID 10c4:ea60 Silicon Labs CP210x UART Bridge
Bus 001 Device 002: ID 046d:c517 Logitech, Inc. LX710 Cordless Desktop Laser
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
rob@NUC5-Rob:~$ ls -al /dev/ttyU* 
crw-rw---- 1 root dialout 188, 0  5 feb 13:08 /dev/ttyUSB0
rob@NUC5-Rob:~$ groups rob
rob : rob tty uucp dialout cdrom floppy sudo audio dip video plugdev netdev bluetooth lpadmin scanner nodegrp
ls -l /dev/serial/by-id
totaal 0
lrwxrwxrwx 1 root root 13  5 feb 13:07 usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_ce24f433c4c9eb11bc9e914f1d69213e-if00-port0 -> ../../ttyUSB0
I will install zigbee2mqtt now

**Comment by tonbor** [comment]
Ok will in 2 hours

**Reply by tonbor** [reply]
Same problem
npm start

> zigbee2mqtt@1.30.0 start
> node index.js

Zigbee2MQTT:info  2023-02-05 13:31:41: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2023-02-05.13-31-41' filename: log.txt
Zigbee2MQTT:info  2023-02-05 13:31:41: Starting Zigbee2MQTT version 1.30.0 (commit #8f781db)
Zigbee2MQTT:info  2023-02-05 13:31:41: Starting zigbee-herdsman (0.14.89)
Zigbee2MQTT:error 2023-02-05 13:31:42: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2023-02-05 13:31:42: Failed to start zigbee
Zigbee2MQTT:error 2023-02-05 13:31:42: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2023-02-05 13:31:42: Exiting...
Zigbee2MQTT:error 2023-02-05 13:31:42: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
    at SerialPort.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/znp/znp.ts:146:28)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/@serialport/stream/dist/index.js:76:22)
    at /opt/zigbee2mqtt/node_modules/@serialport/stream/dist/index.js:112:18

**Reply by tonbor** [reply]
ob@NUC5-Rob:/opt/zigbee2mqtt$ zigbee : zigbee tty dialout
bash: zigbee: opdracht niet gevonden -> not found.......

**Reply by lux73** [reply]
anything is blocking your z2m to accessing the USB port - it's hard to say what it is...

i have found this: https://github.com/Koenkk/zigbee2mqtt/issues/9948

but this for sure will not help - any other Software who is running on your machine?

**Reply by lux73** [reply]
if you are running HomeAssistant make sure the ZHA Addon is deinstalled

if any other SmartHome Software is running make even sure there is no add-on or Programm that is accessing the USB Port

**Reply by tonbor** [reply]
No HA, nothing active

**Reply by lux73** [reply]
anything what i asked and you posted seems good to me

only thing thats strange is your User Groups:

```
rob@NUC5-Rob:~$ groups rob
rob : rob tty uucp dialout cdrom floppy sudo audio dip video plugdev netdev bluetooth lpadmin scanner nodegrp
```

afaik for sure so much groups are not default - but i don't know if you have any Desktop installed - i prefer antime a clean small "netinstallation"

anyhow i'm running out of ideas for now

**Reply by tonbor** [reply]
Yes a desktop

**Comment by lux73** [comment]
last thaught - are you aware that the configuration.yaml is very case sensitive?

one space too much in order will prevent z2m from starting

please post your config with "code" "<>" Tags

**Reply by tonbor** [reply]
I know, I run in it, but solved

**Comment by lux73** [comment]
have you worked out all suggestions from z2m Website if it failes to start?

deinstall ModemManager -> https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#modemmanager-is-installed

`sudo apt-get purge modemmanager`

and disable hciuart? -> https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#hciuart-is-running

`sudo systemctl disable hciuart`

like your Groups there is many Software installed who is likely one that preventing z2m to accessing the USB Port

**Comment by tonbor** [comment]
npm is a long time busy and now is the error:Error: Coordinator failed to start, probably the panID is already in use, try a different panID or channel, pfffffff

**Reply by lux73** [reply]
configuration.yaml:

```
advanced:
  pan_id: 9876
```

change the PanID to any other value as before - don't know why but now it seems the Coordinator is reachable but now it's detectetd the same PanID is using from another Device. Thats not good. So change the PanID

**Reply by tonbor** [reply]
the Panid is auto generated

**Reply by lux73** [reply]
then change it and use a fixed PanID

**Comment by tonbor** [comment]
Zigbee2MQTT:info  2023-02-05 16:16:01: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2023-02-05.16-16-01' filename: log.txt
Zigbee2MQTT:info  2023-02-05 16:16:01: Starting Zigbee2MQTT version 1.30.0 (commit #8f781db)
Zigbee2MQTT:info  2023-02-05 16:16:01: Starting zigbee-herdsman (0.14.89)
Zigbee2MQTT:info  2023-02-05 16:16:03: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2023-02-05 16:16:03: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20211217,"transportrev":2},"type":"zStack3x0"}'
Zigbee2MQTT:info  2023-02-05 16:16:03: Currently 0 devices are joined:
Zigbee2MQTT:warn  2023-02-05 16:16:03: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2023-02-05 16:16:03: Allowing new devices to join.
Zigbee2MQTT:warn  2023-02-05 16:16:03: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2023-02-05 16:16:03: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2023-02-05 16:16:03: Connecting to MQTT server at mqtt://localhost
Zigbee2MQTT:info  2023-02-05 16:16:03: Connected to MQTT server
Zigbee2MQTT:info  2023-02-05 16:16:03: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload '{"state":"online"}'
Zigbee2MQTT:info  2023-02-05 16:16:03: Started frontend on port 0.0.0.0:8080
Zigbee2MQTT:info  2023-02-05 16:16:03: Zigbee2MQTT started!

**Comment by tonbor** [comment]
Changing to fixed PanID

**Reply by lux73** [reply]
nice

**Comment by lux73** [comment]
so deinstall ModemManager and disable hciuart was successfull?

**Reply by tonbor** [reply]
did not

**Comment by tonbor** [comment]
Thanks lux73 for helping me out. Two problems kept me from using Zigbee2Mqtt, first using the port. Problem was   3x, zigbee2mqtt using the port while starting npm, zigbee for domoticz using the port, and 3th no access by local user.
Other what I still not understand the Pan_id auto generated as discribed in the install manual, a static pan_id: 9876,  I heard here for the first time.

**Comment by tonbor** [comment]
I switched to Ubuntu because debian 11 is not very stable and trustworthy I noticed. So all the things I learned that would go wrong above happend with Ubuntu too, ttyUSB0 to user, and PanId static. So no problem to have npm start running with no error in minutes. 
But of course at the end I got stuck. # Create a systemctl configuration file for Zigbee2MQTT
sudo nano /etc/systemd/system/zigbee2mqtt.service, I only did changed user from pi to rob. Npm start worked out of the box every time, but # Start Zigbee2MQTT sudo systemctl start zigbee2mqtt crashed within seconds tells # Show status systemctl status zigbee2mqtt.service...... Replace ExecStart=/usr/bin/npm start with ExecStart=/usr/local/bin/npm start. did the trick

**Comment by darkssside** [comment]
How did you fix the fact that the port was busy?
